### PR TITLE
Add database for tests

### PR DESCRIPTION
**Pull request by tjsharp1**
I think this should allow us to add all-features. The postgres tests just need a pg backend to talk to.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -94,6 +96,39 @@ jobs:
...
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+      - run: cargo install diesel_cli
+      - run: |
+          cd schema
+          diesel migration run
+          cargo fmt
+      - uses: Swatinem/rust-cache@v1
```
Comment (mitchmindtree) [inline .github/workflows/ci.yml:123]: I think if we put this cache step right before the `cargo install diesel_cli` step, it might also cache diesel too.

**Review by mitchmindtree — COMMENTED** [review]
Nice one! Don't forget to add to the `needs` list of `validation-complete` job too

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tjsharp1) [reply .github/workflows/ci.yml]: ok, I moved it up. Looks like it works though, tests are passing.

Comment (mitchmindtree) [reply .github/workflows/ci.yml]: Nice :+1: just need to add the new job to the list of `needs` under the `validation-complete` job now.

**Review by ra0x3 — APPROVED** [review]

**Review by mitchmindtree — APPROVED** [review]
:ok_hand:

**Pull request merged**